test/test/kernel/KernelTests.java

```diff
@@ -19,15 +25,13 @@ public void kernel(TestProcess p) throws Exception {
         Output out = p.profile("-e cpu -d 3 -i 1ms -o collapsed");
         Output err = p.readFile(TestProcess.PROFERR);
         assert out.contains("test/kernel/ListFiles.listFiles;java/io/File");
-        if (!err.contains("Kernel symbols are unavailable")) {
-            assert out.contains("sys_getdents");
-        }
+        assert err.contains("Kernel symbols are unavailable") || out.contains("sys_getdents");
 
         out = p.profile("stop -o flamegraph");
-//        assert out.contains("f\\(\\d+,\\d+,\\d+,\\d,'java/io/File.list'(,\\d+,\\d+,\\d+)?\\)");
-//        if (!err.contains("Kernel symbols are unavailable")) {
-//            assert out.contains("sys_getdents");
-//        }
+        String convertedOut = out.convert(
+                FLAMEGRAPH, COLLAPSED, new ByteArrayInputStream(out.toString().getBytes(StandardCharsets.UTF_8)));
```

Comment:
This non-trivial line is repeated in at least two places. A good indicator to modify the API.

Reply:
Resolved as part of simplification.